### ISSUE TITLE
Cannot create S2I component using name:version format for component type

### ISSUE THREAD

**Issue by dgolovin** [openshift/odo#5144]
/kind bug

## What versions of software are you using?

**Operating System: Linux**

**Output of `odo version`: odo v2.4.0 (b0b2480dc)**

## How did you run odo exactly?

```
odo create nodejs:latest nodejs-ex-3 --context=/home/user/temp/nodejs-ex-3 --app=app1 --project=p1
```
## Actual behavior

```
Devfile Object Validation
 ✗  Checking devfile existence [69471ns]
 ✗  devfile component type "nodejs:latest" is not supported, please run `odo catalog list components` for a list of supported devfile component types
```

## Expected behavior

Devfile component created out of s2i nodejs:latest

## Any logs, error output, etc?

See above

**Comment by dgolovin** [comment]
How do I create S2I component with 2.4.0 if `--s2i` flag is removed?

**Comment by kadel** [comment]
I'm afraid that you are not going to like my answer :-/ 
In 2.3.1 we added a deprecation warring to `--s2i` flag with information that it will be removed in future versions. (https://github.com/openshift/odo/issues/4932)
Now in 2.4.0 it got removed.

We intentionally did a good "really good job" hiding this in release notes under "Cleanup/Refactor". (cc @dharmit)

I'm sorry about this it should have been better communicated.
